x-pack/plugins/maps/public/shared/layers/sources/es_search_source/create_source_editor.js

```diff
@@ -52,9 +61,28 @@ export class CreateSourceEditor extends Component {
       isLoadingIndexPattern: true,
```

Comment:
defaults here are getting configured in multiple places.

I'd extract the reset state in single place 

RESET_STATE = {
    indexPatternId: '',
    geoField: '',
    noGeoIndexPatternsExist: false,
    indexDocCount: DOC_COUNT_UNSET,
    filterByMapBounds: DEFAULT_FILTER_BY_MAP_BOUNDS,
    showFilterByBoundsSwitch: false
  };

and mix it in here
https://github.com/elastic/kibana/blob/853235455ff9458d182bb754540f1f1d51a75cb9/x-pack/plugins/maps/public/shared/layers/sources/es_search_source/create_source_editor.js#L34-L42

and here.

https://github.com/elastic/kibana/blob/853235455ff9458d182bb754540f1f1d51a75cb9/x-pack/plugins/maps/public/shared/layers/sources/es_search_source/create_source_editor.js#L60-L68